tiledb/sm/query/reader.cc

```diff
@@ -3161,6 +3168,43 @@ Status Reader::copy_attribute_values(
   STATS_END_TIMER(stats::Stats::TimerType::READ_COPY_ATTR_VALUES);
 }
 
+Status Reader::add_extra_offset() {
+  for (const auto& it : buffers_) {
+    const auto& name = it.first;
+    // TODO: keep that check?
+    if (read_state_.overflowed_)
+      break;
```

Comment:
this can be removed, contingent on addressing my comment about moving the call to this routine from `sparse_read` to `read`.